core/thread_flags.c

```diff
@@ -80,7 +80,8 @@ thread_flags_t thread_flags_wait_one(thread_flags_t mask)
     _thread_flags_wait_any(mask);
     thread_t *me = (thread_t*) sched_active_thread;
     unsigned tmp = me->flags & mask;
-    return _thread_flags_clear_atomic(me, thread_flags_clear(1 << bitarithm_lsb(tmp)));
+    tmp &= -tmp;
```

Comment:
Better `~tmp + 1` so it does not break for architectures that don't use two's complement. At least in theory there could be such systems. Even with `-O0` gcc understands and optimizes the code, so there is no drawback.